pkg/cleaner/handler.go

```diff
@@ -0,0 +1,60 @@
+package cleaner
```

Comment:
1: avoid generic namesl ike "handler"
2: framework could go into its own package
3: this really sohuld be an interface IMO
4: separate the IPAM cleaner imlementations from the interface itself

Reply:
1. renamed to interface.go, but that is also generic. Why generic names a concern?
2. moved to `ipamfree` package
3. Still working on it
4. separated calico implementation into `ipamfree/calico.go`

Reply:
About the 3.:
By interface do you mean golang interface? I would like to wait for at least one more static CNI implementation come in to picture before migrating to an interface.

Reply:
yep. k

Reply:
I refactored code to use a DeleteDanmEp Interface for different implementations. Added danm and calico implementations.
I did not want to differenciate between Danm's implementation and other static CNI implementation to make Interface generic.
This cleaner API works the following way:

1. First it is always uses DANM's implementation to determine if DanmEp's IPAM is managed by it, then if :
  1. IPAM is managed by DANM, it invokes DANM's DeleteDanmEp function with underlying DANM IPAM cleanup
  2. IPAM is NOT managed by DANM, it moves on to the next DeleteDanmEp implementation (calico is the only one currently)
2. It tries the second implementation to determine if DanmEp's IPAM is managed by it,
... etc.
N. If no implementation have been found sufficcient, then emit error

Reply:
I eventually separated IPv4 and IPv6 handling